src/promql/src/planner.rs

```diff
@@ -1351,6 +1377,107 @@ impl PromPlanner {
             .context(DataFusionPlanningSnafu)
     }
 
+    fn set_op_on_non_field_columns(
+        &self,
+        left: LogicalPlan,
+        right: LogicalPlan,
+        left_tag_cols: Vec<String>,
+        right_tag_cols: Vec<String>,
+        op: TokenType,
+        modifier: &Option<BinModifier>,
+    ) -> Result<LogicalPlan> {
+        let mut left_tag_col_set = left_tag_cols.into_iter().collect::<HashSet<_>>();
+        let mut right_tag_col_set = right_tag_cols.into_iter().collect::<HashSet<_>>();
+
+        // apply modifier
+        if let Some(modifier) = modifier {
+            // one-to-many and many-to-one are not supported
```

Comment:
Will they be supported in the future?  If so, maybe change it to `TODO` item. If not, why?

Reply:
This limitation is from PromQL. `group_left` and `group_right`:
>Grouping modifiers can only be used for [comparison](https://prometheus.io/docs/prometheus/latest/querying/operators/#comparison-binary-operators) and [arithmetic](https://prometheus.io/docs/prometheus/latest/querying/operators/#arithmetic-binary-operators). Operations as and, unless and or operations match with all possible entries in the right vector by default.

https://prometheus.io/docs/prometheus/latest/querying/operators/#group-modifiers

Reply:
```
any-to-one and one-to-many matching are advanced use cases that should be carefully considered. Often a proper use of ignoring(<labels>) provides the desired outcome.
```

Looks like it supports but doesn't recommend using them.

Reply:
That's another context. For set operation:
>Grouping modifiers can only be used for [comparison](https://prometheus.io/docs/prometheus/latest/querying/operators/#comparison-binary-operators) and [arithmetic](https://prometheus.io/docs/prometheus/latest/querying/operators/#arithmetic-binary-operators).